lib/rule-tester/rule-tester.js

```diff
@@ -1160,6 +1174,10 @@ class RuleTester {
 						});
 
 						if (hasOwnProperty(error, "message")) {
+							assert.ok(
+								requireMessage !== "messageId",
+								"The test run requires 'messageId' but the error (also) uses 'message'.",
+							);
```

Comment:
This "also" might be confusing as we don't know at this point if `messageId` is present. Perhaps we could move this check after the next one (`!hasOwnProperty(error, "messageId")`) and remove "also"?